pkg/tests/tasks/security/certificate/external_ca_test.go

```diff
@@ -16,101 +16,116 @@ package certificate
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"testing"
-	"time"
 
-	"github.com/maistra/maistra-test-tool/pkg/examples"
+	"github.com/maistra/maistra-test-tool/pkg/app"
 	"github.com/maistra/maistra-test-tool/pkg/util"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/curl"
 	"github.com/maistra/maistra-test-tool/pkg/util/env"
-	"github.com/maistra/maistra-test-tool/pkg/util/log"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
 	"github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
-func cleanupExternalCert() {
-	log.Log.Info("Cleanup")
+func TestExternalCertificate(t *testing.T) {
+	test.NewTest(t).Id("T17").Groups(test.Full, test.ARM, test.InterOp).Run(func(t test.TestHelper) {
 
-	bookinfo := examples.Bookinfo{Namespace: "bookinfo"}
-	bookinfo.Uninstall()
+		hack.DisableLogrusForThisTest(t)
 
-	util.Shell(`kubectl -n %s delete secret cacerts`, meshNamespace)
-	util.Shell(`kubectl -n %s patch smcp/%s --type=json -p='[{"op": "remove", "path": "/spec/security/certificateAuthority"}, {"op": "remove", "path": "/spec/security/dataPlane"}]'`, meshNamespace, smcpName)
-	util.Shell(`oc -n %s wait --for condition=Ready smcp/%s --timeout 180s`, meshNamespace, smcpName)
-	time.Sleep(time.Duration(60) * time.Second)
-}
+		t.Log("Testing external certificates.")
+
+		var tmpDir = ""
 
-func TestExternalCert(t *testing.T) {
-	test.NewTest(t).Id("T17").Groups(test.Full, test.ARM, test.InterOp).NotRefactoredYet()
+		t.Cleanup(func() {
 
-	defer cleanupExternalCert()
+			if tmpDir != "" {
+				t.Log("Removing temporary directory")
+				os.RemoveAll(tmpDir)
+			}
 
-	log.Log.Info("Test External Certificates")
-	log.Log.Info("Enable Control Plane MTLS")
+			t.Log("Recreating bookinfo namespace.")
+			oc.RecreateNamespace(t, "bookinfo")
 
-	t.Run("Security_plugging_external_cert_test", func(t *testing.T) {
-		defer util.RecoverPanic(t)
+			t.Log("Undoing SMCP patch.")
+			oc.Patch(t, meshNamespace, "smcp", smcpName, "json", CertSMCPPathRemovalPatch)
+			oc.WaitSMCPReady(t, meshNamespace, smcpName)
+		})
 
-		log.Log.Info("Adding an external CA")
-		util.Shell(`kubectl create -n %s secret generic cacerts --from-file=%s --from-file=%s --from-file=%s --from-file=%s`,
+		t.LogStep("Adding external CA.")
+		oc.DeleteSecret(t, meshNamespace, "cacerts")
+		shell.Executef(t,
```

Comment:
We could use a more flexible version of `oc.CreateTLSSecretWithCACert` to deal with this potentially.